Core/Core/Inbox/AttachmentPicker/ViewModel/AudioPickerViewModel.swift

```diff
@@ -0,0 +1,248 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import AVFoundation
+import Combine
+import CombineExt
+
+class AudioPickerViewModel: NSObject, ObservableObject, AVAudioPlayerDelegate {
+
+    // MARK: Private
+    private var audioRecorder: AVAudioRecorder!
+    private var audioPlayer: AVAudioPlayer!
+    private var timer: Timer!
+    private let formatter: DateComponentsFormatter
+    private let router: Router
+    private var subscriptions = Set<AnyCancellable>()
+    private var onSelect: (URL) -> Void
+
+    // MARK: Outputs
+
+    @Published public var isRecording: Bool = false
+    @Published public var isRecorderLoading: Bool = false
+    @Published public var loadingAnimationRotation = 0.0
+    @Published public var isPlaying: Bool = false
+    @Published public var isReplay: Bool = false
+    @Published public var url: URL!
+    @Published public var recordingLengthString: String = ""
+    @Published public var audioPlayerPositionString: String = ""
+    @Published public var audioPlayerPosition: Double = 0
+    @Published public var audioPlayerDurationString: String = ""
+    @Published public var audioPlotDataSet: [AudioPlotData] = []
```

Comment:
I think we shouldn't let the outside world modify most of these properties. If we define these properties like `@Published public private(set) var` we would allow reading from anywhere, but limit writing to the view model itself.